cmd/clusterctl/examples/aws/BUILD.bazel

```diff
@@ -0,0 +1,98 @@
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
+
+genrule(
+    name = "provider-components-base",
+    srcs = [
+        "//config:aws_provider.yaml",
+        "provider-components-base/kustomization.yaml",
+        "provider-components-base/stateful-set-credentials-patch.yaml",
+        "//vendor/sigs.k8s.io/cluster-api/config:cluster-api-yaml",
+        "//cmd/manager:manager-version-patch.yaml",
+    ],
+    outs = ["provider-components-base.yaml"],
+    cmd = """cp -f {kustomization} $(@D) && \\
+             cp -f {stateful_set_credentials_patch} $(@D) && \\
+             cp -f {stateful_set_image_patch} $(@D) && \\
+             cp -f {base_provider} $(@D)/aws-provider-base.yaml && \\
+             cp -f {cluster_api_base} $(@D)/cluster-api-base.yaml && \\
+             {kustomize} build $(@D) > $@
+    """.format(
+        kustomize = "$(location @io_k8s_sigs_kustomize//:kustomize)",
```

Comment:
Organizing these by either the same order they appear in the cmd string or alphabetically might be nice, but since none are reused I'd rather see .format on each line with the join syntax. My goals with this suggestion are keeping each command isolated and therefore easier to spot when a variable is not used in the format syntax.

Reply:
I'll change to this type then.